src/main/java/com/minecolonies/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -828,9 +828,9 @@ private static ItemStack handleKeepX(@NotNull Map<ItemStorage, Integer> keptX,
      */
     private static boolean keptEnough(@NotNull Map<ItemStorage, Integer> kept, @NotNull Map<ItemStorage, Integer> keep, @NotNull ItemStack stack)
```

Comment:
My TLDR at this point would be that I think we should change the data structure so that we aren't iterating through a HashMap. In a PR about improving performance, I think the changes made to this method were negligible, and I think we can improve it a lot.